tsdb/db_test.go

```diff
@@ -1497,7 +1469,8 @@ func TestTimeRetention(t *testing.T) {
 	}()
 
 	blocks := []*BlockMeta{
-		{MinTime: 500, MaxTime: 900}, // Oldest block
+		{MinTime: 500, MaxTime: 900},  // Oldest block
+		{MinTime: 500, MaxTime: 1000}, // Coinciding exactly with the retention duration.
```

Comment:
I'm afraid, we no longer test the `>` case now.
I was thinking about having two testCases in here, with the existing

```
[]*BlockMeta{
{MinTime: 500, MaxTime: 900}, // Oldest block
{MinTime: 1000, MaxTime: 1500},
{MinTime: 1500, MaxTime: 2000},
}
```

to test the `>` case.

And add

```
[]*BlockMeta{
{MinTime: 500, MaxTime: 1000}, // Oldest block
{MinTime: 1000, MaxTime: 1500},
{MinTime: 1500, MaxTime: 2000},
}
```

to test the `=` case.

Reply:
Sounds good, on it.

Reply:
(the existing checks should pass for both of them without any tweaks, we just need to add the loop)

Reply:
@machine424 I split `TestTimeRetention` into two sub-tests, one for each case (I verified that they break if I modify the retention check). PTAL.